### PR TITLE
Add baseline tests to CI tests

### PR DESCRIPTION
**Pull request by priyakasimbeg**
Add test_baselines.py to CI workflow, for 12 enabled baselines this adds about 4:37 minutes to the CI tests.
Also fix bug in mnist input_pipeline.

### CODE REVIEW

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -181,4 +181,6 @@ jobs:
         pytest -vx tests/test_num_params.py
         pytest -vx tests/test_param_shapes.py
         pytest -vx tests/test_param_types.py
-        pytest -vx tests/test_ssim.py
\ No newline at end of file
+        pytest -vx tests/test_ssim.py
+        pytest -vx tests/submission_runner_test.py
+        pytest -vx tests/test_baselines.py
```
Comment (znado) [inline .github/workflows/CI.yml:186]: will this download the MNIST data? that'll be a bit slow but probably is OK

File: .github/workflows/CI.yml [hunk 2]
```diff
@@ -181,4 +181,6 @@ jobs:
         pytest -vx tests/test_num_params.py
         pytest -vx tests/test_param_shapes.py
         pytest -vx tests/test_param_types.py
-        pytest -vx tests/test_ssim.py
\ No newline at end of file
+        pytest -vx tests/test_ssim.py
+        pytest -vx tests/submission_runner_test.py
```
Comment (znado) [inline .github/workflows/CI.yml:185]: do we even need the `test_submission` function in this file anymore? it seems like that's covered by test_baselines.py now

**Review by znado — CHANGES_REQUESTED** [review]

File: .github/workflows/CI.yml (hunk 2 above)
Comment (priyakasimbeg) [reply .github/workflows/CI.yml]: I suppose we don't I removed it.

File: .github/workflows/CI.yml (hunk 1 above)
Comment (priyakasimbeg) [reply .github/workflows/CI.yml]: It took about 4 minutes 37 seconds to run all 12 baseline tests I enabled so I think it is ok too.

**Review by znado — APPROVED** [review]

**Pull request merged**